src/shell.rs

```diff
@@ -364,19 +396,19 @@ impl Default for MessageInfo {
 
 impl From<ColorChoice> for MessageInfo {
     fn from(color_choice: ColorChoice) -> MessageInfo {
-        MessageInfo::new(color_choice, Verbosity::Normal)
+        MessageInfo::new(color_choice, Verbosity::Normal).cross_debug_check()
     }
 }
 
 impl From<Verbosity> for MessageInfo {
     fn from(verbosity: Verbosity) -> MessageInfo {
-        MessageInfo::new(ColorChoice::Auto, verbosity)
+        MessageInfo::new(ColorChoice::Auto, verbosity).cross_debug_check()
     }
 }
 
 impl From<(ColorChoice, Verbosity)> for MessageInfo {
     fn from(info: (ColorChoice, Verbosity)) -> MessageInfo {
-        MessageInfo::new(info.0, info.1)
+        MessageInfo::new(info.0, info.1).cross_debug_check()
```

Comment:
is this really needed? I added it just to be safe and consistent

Reply:
I think we can move this into the `new` and `create` constructors, probably with `new` requiring the `cross_debug` bool, and `create` parsing the value. Maybe there should probably be a static function to detect it.